recipes/liquid-dsp/0003-Set-library-sonames-properly-to-allow-for-splitting-.patch

```diff
@@ -0,0 +1,125 @@
+@@ -265,16 +268,29 @@ darwin*)
+     AC_DEFUN([AC_PROG_LIBTOOL], [AC_CHECK_TOOL(LIBTOOL, libtool, :)])
+     AC_PROG_LIBTOOL
+ 
++    SONAME=libliquid.$SOVERSION.dylib
+     AR_LIB=ar	    # archive extension
+     SH_LIB=dylib    # shared library extension
+     REBIND=""
+     ;;
++mingw*)
++    AN_MAKEVAR([AR], [AC_PROG_AR])
++    AN_PROGRAM([ar], [AC_PROG_AR])
++    AC_DEFUN([AC_PROG_AR], [AC_CHECK_TOOL(AR, ar, :)])
++    AC_PROG_AR
++
++    SONAME=liquid-$SOVERSION.dll
++    AR_LIB=a	    # archive extension
++    SH_LIB=dll	    # shared library extension
++    REBIND=ldconfig
++    ;;
+ *)
+     AN_MAKEVAR([AR], [AC_PROG_AR])
+     AN_PROGRAM([ar], [AC_PROG_AR])
+     AC_DEFUN([AC_PROG_AR], [AC_CHECK_TOOL(AR, ar, :)])
+     AC_PROG_AR
+ 
++    SONAME=libliquid.so.$SOVERSION
+     AR_LIB=a	    # archive extension
+     SH_LIB=so	    # shared library extension
+     REBIND=ldconfig
+@@ -288,6 +304,7 @@ AC_SUBST(LIBS)                      # shared libraries (-lc, -lm, etc.)
+ AC_SUBST(MLIBS_DOTPROD)             # 
+ AC_SUBST(MLIBS_VECTOR)              #
+ 
++AC_SUBST(SONAME)
+ AC_SUBST(AR_LIB)                    # archive library
+ AC_SUBST(SH_LIB)                    # output shared library target
+ AC_SUBST(REBIND)                    # rebinding tool (e.g. ldconfig)
+diff --git a/makefile.in b/makefile.in
+index 09416d87..542087e9 100644
+--- a/makefile.in
++++ b/makefile.in
+@@ -44,6 +44,7 @@
+ # autoconf initialization macros
+ NAME		:= @PACKAGE_NAME@
+ VERSION		:= @PACKAGE_VERSION@
++SONAME		:= @SONAME@
+ BUGREPORT	:= @PACKAGE_BUGREPORT@
+ AR_LIB		:= @AR_LIB@
+ SH_LIB		:= @SH_LIB@
+@@ -1244,7 +1245,7 @@ benchmark_sources :=						\
+ 
+ # Shared library targets
+ ARCHIVE_LIB	= libliquid.${AR_LIB}
+-SHARED_LIB	= libliquid.${SH_LIB}
++SHARED_LIB	= ${SONAME}
+ 
+ #
+ # darwin
+@@ -1253,7 +1254,7 @@ libliquid.ar : $(objects)
+ 	${LIBTOOL} -static -o $@ $^
+ 
+ # gcc -dynamiclib -install_name libliquid.dylib -o libliquid.dylib libmodem.a libutility.a 
+-libliquid.dylib : $(objects)
++libliquid%.dylib : $(objects)
+ 	$(CC) -dynamiclib ${COVERAGE_FLAGS} -install_name $@ -o $@ $^ $(LDFLAGS) $(LIBS)
+ 
+ #
+@@ -1263,9 +1264,15 @@ libliquid.a : $(objects)
+ 	${AR} r $@ $^
+ 	${RANLIB} $@
+ 
+-libliquid.so : libliquid.a
++libliquid.so% : libliquid.a
+ 	$(CC) $(CFLAGS) $(LDFLAGS) -shared -Xlinker -soname=$@ -o $@ -Wl,-whole-archive $^ -Wl,-no-whole-archive $(LIBS)
+ 
++#
++# mingw
++#
++liquid%.dll: libliquid.a
++	$(CC) $(CFLAGS) $(LDFLAGS) -shared -Xlinker -soname=$@ -o $@ -Wl,-whole-archive $^ -Wl,-no-whole-archive -Wl,--output-def,liquid.def -Wl,--out-implib,$@.a $(LIBS)
++
+ # static archive and library objects
+ all: ${ARCHIVE_LIB} ${SHARED_LIB}
+ 
+@@ -1288,10 +1295,9 @@ install: all
+ 	mkdir -p $(DESTDIR)$(prefix)/include/liquid
+ 	mkdir -p $(DESTDIR)$(libdir)
+ 	install -m 644 -p libliquid.${AR_LIB} $(DESTDIR)$(libdir)/libliquid.${AR_LIB}.${VERSION}
+-	install -m 644 -p libliquid.${SH_LIB} $(DESTDIR)$(libdir)/libliquid.${SH_LIB}.${VERSION}
```

Comment:
> Splitting liquid-dsp into multiple outputs, while I like having versioned lib packages like libliquid1, was also convenient because the base build requires both compiler('m2w64_c') (for the actual build) and compiler('c') (to have MSVC for creating a compatible import lib) and the outputs allow for easily avoiding the compiler('c') dependencies being added to the package.

I'm a bit confused. Is the SO name from upstream or something you added? If it is something you added or tweaked. Please open a PR upstream for this patch so that our artifacts are as close to canonical as possible.

Reply:
It took a while to sort this out, so confusion is understandable. The SONAME that results from this patch is still the one used upstream (e.g. `libliquid.so.1`), except upstream just makes copies/links to have that as the output filename but doesn't follow the proper incantation to use that as the internal SONAME. The result is that the library is secretly unversioned while appearing to be versioned. This patch fixes that so that it is actually versioned and works that way on all 3 platforms. I'll certainly try to get this upstream, which will be easier now that I have proof that it works!

Reply:
Thanks for contributing upstream. Carrying patches in a feedstock is a pain.